pkg/kubectl/resource_printer.go

```diff
@@ -350,106 +349,61 @@ func podHostString(host, ip string) string {
+		return fmt.Sprintf("%dh", hours)
+	} else if hours < 24*364 {
+		return fmt.Sprintf("%dd", hours/24)
+	}
+	return fmt.Sprintf("%dy", int(d.Hours()/24/365))
+}
+
 // translateTimestamp returns the elapsed time since timestamp in
 // human-readable approximation.
 func translateTimestamp(timestamp util.Time) string {
-	return units.HumanDuration(time.Now().Sub(timestamp.Time))
-}
-
-// interpretContainerStatus interprets the container status and returns strings
-// associated with columns "STATUS", "CREATED", and "MESSAGE".
-// The meaning of MESSAGE varies based on the context of STATUS:
-//     STATUS: Waiting; MESSAGE: reason for waiting
-//     STATUS: Running; MESSAGE: reason for the last termination
-//     STATUS: Terminated; MESSAGE: reason for this termination
-func interpretContainerStatus(status *api.ContainerStatus) (string, string, string, error) {
-	// Helper function to compose a meaning message from terminate state.
-	getTermMsg := func(state *api.ContainerStateTerminated) string {
-		var message string
-		if state != nil {
-			message = fmt.Sprintf("exit code %d", state.ExitCode)
-			if state.Reason != "" {
-				message = fmt.Sprintf("%s, reason: %s", message, state.Reason)
-			}
-		}
-		return message
-	}
-
-	state := &status.State
-	if state.Waiting != nil {
-		return "Waiting", "", state.Waiting.Reason, nil
-	} else if state.Running != nil {
-		// Get the information of the last termination state. This is useful if
-		// a container is stuck in a crash loop.
-		message := getTermMsg(status.LastTerminationState.Terminated)
-		if message != "" {
-			message = "last termination: " + message
-		}
-		stateMsg := "Running"
-		if !status.Ready {
-			stateMsg = stateMsg + " *not ready*"
-		}
-		return stateMsg, translateTimestamp(state.Running.StartedAt), message, nil
-	} else if state.Terminated != nil {
-		return "Terminated", translateTimestamp(state.Terminated.StartedAt), getTermMsg(state.Terminated), nil
-	}
-	return "", "", "", fmt.Errorf("unknown container state %#v", *state)
+	return shortHumanDuration(time.Now().Sub(timestamp.Time))
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool) error {
-	var name string
+	name := pod.Name
 	if withNamespace {
-		name = types.NamespacedName{pod.Namespace, pod.Name}.String()
-	} else {
-		name = pod.Name
+		name = types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}.String()
 	}
 
-	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
-		name,
-		pod.Status.PodIP,
-		"", "",
-		podHostString(pod.Spec.NodeName, pod.Status.HostIP),
-		formatLabels(pod.Labels),
-		pod.Status.Phase,
-		translateTimestamp(pod.CreationTimestamp),
-		pod.Status.Message,
-	)
-	if err != nil {
-		return err
+	restarts := 0
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+	reason := string(pod.Status.Phase)
+
+	// TODO: replace pod.Status.Message with pod.Status.Reason (to be added soon)
+	if pod.Status.Message != "" {
```

Comment:
If you want to defer this to a subsequent PR, just don't initialize reason to message right now.